src/frontend/src/optimizer/plan_node/logical_scan.rs

```diff
@@ -223,28 +223,34 @@ impl LogicalScan {
         &self.predicate
     }
 
+    /// get the Mapping of columnIndex from internal column index to output column index
+    pub fn i2o_col_mapping(&self) -> ColIndexMapping {
+        ColIndexMapping::with_remaining_columns(
+            &self.output_col_idx,
+            self.table_desc().columns.len(),
+        )
+    }
+
     /// Return indices of fields the output is ordered by and
     /// corresponding direction
     pub fn get_out_column_index_order(&self) -> Order {
-        let id_to_op_idx = Self::get_id_to_op_idx_mapping(&self.output_col_idx, &self.table_desc);
-        Order::new(
+        let id_to_tb_idx = self.table_desc.get_id_to_op_idx_mapping();
```

Comment:
Is `Self::get_id_to_op_idx_mapping` still useful, and what's their difference? Maybe add some comments

Reply:
I think they are both useful.

Reply:
here we need an internal order before `output_indices` to use the `ColumnIndexMapping::rewrite_provide_order`

Reply:
I think we can have `self.table_desc.order()`

Reply:
a little strange to me and I think this is the BatchScan's behavior

Reply:
and for the `get_id_to_op_idx_mapping` should also be used on distribution and when we fix https://github.com/singularity-data/risingwave/blob/main/src/frontend/src/optimizer/plan_node/batch_seq_scan.rs#L96 later.